### PR TITLE
Implemented: shopify CDN compatible image component (#102)

### PR DESCRIPTION
**Pull request by k2maan**
Closes: #102 
Implemented Shopify CDN-compatible image component with a configurable size that can be sent as a prop/option in the component.

### CODE REVIEW

File: src/components/ShopifyCDNImage.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  template: `
+            <img :src="imageUrl"/>
+            `,
+  data() {
+    return {
+      imageUrl: ''
+    }
+  },
+  props: ['src', 'size'],
+  async mounted() {
```
Comment (adityasharma7) [inline src/components/ShopifyCDNImage.ts:14]: We are using async without await

File: src/components/ShopifyCDNImage.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    this.setImageUrl();
+  },
+  updated() {
+    this.setImageUrl();
+  },
+  methods: {
+    addSizeInImgUrl(src: string, size: string) {
+      if (src) {
+        // return original size if no size is given
+        if (!size) return src
```
Comment (adityasharma7) [inline src/components/ShopifyCDNImage.ts:24]: Method should have size optional if are checking it here as well

File: src/components/ShopifyCDNImage.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  },
+  props: ['src', 'size'],
+  async mounted() {
+    this.setImageUrl();
+  },
+  updated() {
+    this.setImageUrl();
+  },
+  methods: {
+    addSizeInImgUrl(src: string, size: string) {
```
Comment (adityasharma7) [inline src/components/ShopifyCDNImage.ts:21]:
```suggestion
    prepareImgUrl(src: string, size: string) {
```

File: src/components/ShopifyCDNImage.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+          resolve(true);
+        }
+        img.onerror = function () {
+          reject(false);
+        }
+        img.src = src;
+      })
+    },
+    setImageUrl() {
+      const src = this.addSizeInImgUrl(this.src, this.size)
```
Comment (adityasharma7) [inline src/components/ShopifyCDNImage.ts:46]: For assets or images from the resource server, this is not needed. As the component is specific to Shopify, we could remove the handling for assets as well resource server for now.

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/components/ShopifyCDNImage.ts (hunk 4 above)
Comment (k2maan) [reply src/components/ShopifyCDNImage.ts]: Removed support for resourceUrl

**Pull request merged**